File: app/src/main/java/com/geekapps/rsstestapp/data/error/ErrorUtils.java
package com.geekapps.rsstestapp.data.error;

import org.json.JSONException;
import org.json.JSONObject;

public class ErrorUtils {

    public static String convertError(String json) throws JSONException {
        return json.contains("error_description") ?
                new JSONObject(json).getString("error_description") : json;
    }
}

File: app/src/main/java/com/geekapps/rsstestapp/data/error/ErrorUtils.kt
package com.brilliatnbrains.markchat.data.error

import org.json.JSONException
import org.json.JSONObject

object ErrorUtils {

    @Throws(JSONException::class)
    fun convertError(json: String): String {
        return if (json.contains("error_description")) {
            val jsonObj = JSONObject(json)
            jsonObj.getString("error_description")
        } else {
            json
        }
    }
}